### DISCUSSION TITLE
Possible to get auto import completions without adding import statements?

### DISCUSSION THREAD

**Discussion by user-grinch** [microsoft/pylance-release#2740] [Q&A]
I have python embedded into my program and added a few functions using the Python C API. I'd like to get PyLance IntelliSense providing autocompletion for it.

After setting up paths to a dummy python file (with dummy functions), PyLance seems to suggest auto completions for my functions but also adds the top-level ```from x import y``` statements. I've tried to set the ```python.analysis.autoImportCompletions``` to ```false``` but it disables the autocompletion entirely.

So, how to get the autocompletion without PyLance adding the import statements at the top? (I know it's not practical but I think it suits my use case here)

**Comment by erictraut** [comment]
Pylance is built on Microsoft's open-source static type analyzer called pyright. Pyright has a way to extend the list of built-in symbols (those that are available without explicitly importing them). Here's a link to the [documentation for this feature](https://github.com/microsoft/pyright/blob/main/docs/builtins.md).
